runtime-modules/project-token/src/lib.rs

```diff
@@ -439,6 +444,124 @@ decl_module! {
+        /// no-op if `amount.is_zero()`
+        #[weight = 10_000_000] // TODO: adjust weight
+        fn participate_in_split(
+            origin,
+            token_id: T::TokenId,
+            amount: TokenBalanceOf<T>,
+        ) -> DispatchResult {
+            let sender = ensure_signed(origin)?;
+
+            if amount.is_zero() { return Ok(()) }
+
+            let token_info = Self::ensure_token_exists(token_id)?;
+            let split_info = token_info.revenue_split.ensure_active::<T>()?;
+
+            let current_block = Self::current_block();
+            ensure!(
+                split_info.timeline.is_ongoing(current_block),
+                Error::<T>::RevenueSplitNotOngoing
+            );
+
+            let account_info = Self::ensure_account_data_exists(token_id, &sender)?;
+
+            account_info.ensure_can_stake::<T>(amount)?;
+
+            // it should not really be possible to have supply == 0 with staked amount > 0
+            debug_assert!(!token_info.total_supply.is_zero());
+            let dividend_amount = Self::compute_revenue_split_dividend(
+                amount,
+                token_info.total_supply,
+                split_info.allocation,
+            );
+
+            let treasury_account: T::AccountId = Self::module_treasury_account();
+            Self::ensure_can_transfer_joy(&treasury_account, dividend_amount)?;
+
+            // == MUTATION SAFE ==
+
+            <Joy<T> as Currency<T::AccountId>>::transfer(
+                &treasury_account,
+                &sender,
+                dividend_amount,
+                ExistenceRequirement::KeepAlive,
+            )?;
+
+            TokenInfoById::<T>::mutate(token_id, |token_info| {
+                token_info.revenue_split.account_for_dividend(dividend_amount);
+            });
+
+            AccountInfoByTokenAndAccount::<T>::mutate(token_id, &sender, |account_info| {
+                account_info.stake(token_info.latest_revenue_split_id, amount);
+            });
+
+            Self::deposit_event(RawEvent::UserParticipatedInSplit(
+                token_id,
+                sender,
+                amount,
+                dividend_amount,
+                token_info.latest_revenue_split_id,
+            ));
+
+            Ok(())
+        }
+
+        /// Split-participating user leaves revenue split
+        /// Preconditions
+        /// - `token` must exist for `token_id`
+        /// - `account` must exist for `(token_id, sender)` with `origin` signed by `sender`
+        /// - `account.staking status.is_some()'
+        /// - if `(account.staking_status.split_id == token.latest_revenue_split_id`
+        ///    AND `token.revenue_split` is active) THEN split staking period  must be ended
+        ///
+        /// Postconditions
+        /// - `account.staking_status` set to None
+        #[weight = 10_000_000] // TODO: adjust weight
+        fn abandon_revenue_split(origin, token_id: T::TokenId) -> DispatchResult {
```

Comment:
This name had me long confused, was reading impl and expecting this was about exiting and refunding your released $JOY.

I think at least call it something like `leave_inactive_revenue_split` or something to that effect.

Reply:
This is meant to allow a user to effectively **exit** a revenue split after the staking period is expired:
how about `exit_revenue_split`?

Reply:
hmm, still confusing imho., but its fine.